### PR TITLE
constant-state-variables.rst and mapping-types.rst translated, reference-types.rst translation started, fixed little mistakes in layout-of-source-files.rst, added a public version of this project in readthedocs.org and was noted in README.md

### PR DESCRIPTION
**Pull request by juanjo-suarez**

### CODE REVIEW

**Comment by elijahhampton** [comment]
Hey thanks! Will try to review it this weekend.

**Comment by juanjo-suarez** [comment]
> Hey thanks! Will try to review it this weekend.

I am willing to translate other pages, but first I am waiting for your review to confirm if this style or format is well and useful, before going on the same way. Thx in advance! 😉

**Comment by elijahhampton** [comment]
> > Hey thanks! Will try to review it this weekend.
> 
> I am willing to translate other pages, but first I am waiting for your review to confirm if this style or format is well and useful, before going on the same way. Thx in advance! 😉

Sorry for the delay. I have been busy with work and the holidays. Everything will be reviewed today! Thank you!

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: I don't think this is the best place for such a comment. Please note this is the Spanish translation for the official Solidity documentation. 

If you wish to raise an issue about the clarity of the documentation it might be best to do that in the official repository https://github.com/ethereum/solidity.

Comment (juanjo-suarez) [reply docs/types/mapping-types.rst]: Thanks. I had some doubts about this..., but you're right! I just removed my personal clarification :) If you notice any other detail which should be fixed, let me know. Thx.

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: I believe a more accurate way of expressing this title would be "Mapping Iterable" given the usage of "Iterable" as an adjective in the original title.

If you have another opinion on this I am open to hearing it.

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: Second person -> third person (you, see above comment at line 14)

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: Since we haven't had many translators we have kind of been sticking to an "unspoken rule" of translating using third person formal "Ud." of "you" in translations. If this is not documented in the README we will need to update it.

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: Second person -> third person (you, see above comment at line 14)

File: docs/types/mapping-types.rst
Comment (elijahhampton) [inline docs/types/mapping-types.rst]: Second person -> third person (you, see above comment at line 14)

File: docs/types/mapping-types.rst
Comment (juanjo-suarez) [reply docs/types/mapping-types.rst]: Ops, sorry! I did not note that you had been using the third-person style before. My personal preference for these contexts is the second-person, but the third-person is also absolutely correct. I will write in the third person from now. Of course, I agree with you that is a good idea to set this convention in the README for avoiding incoherence in mixed-use.

File: docs/types/mapping-types.rst
Comment (juanjo-suarez) [reply docs/types/mapping-types.rst]: I understand your point of view. Both are grammatically correct with exactly the same meaning :) Probably, one is more natural (at least, for me), while another is more literal (this is obvious). But, don't worry, in my personal opinion, if we are in doubt, literal is first. No problem.

**Review by elijahhampton — APPROVED** [review]

**Comment by elijahhampton** [comment]
@juanjo-suarez Are you able to merge this MR? I'm not sure if everyone has the ability once approved. If not, I will merge it now.

**Comment by juanjo-suarez** [comment]
Hey @elijahhampton! No, I cannot merge it for my own. I thought that maybe more than one approval was required, and I was waiting for that.

**Pull request merged**
